### DISCUSSION TITLE
Login with SSO - React native

### DISCUSSION THREAD

**Discussion by HeinTouchableFR** [directus/directus#11080] [Archive]
Hello,
I am developing a mobile application with react-native and I would like to be able to offer customers to connect via SSO. I have made the configuration to be able to connect via google and it works correctly for web applications. However, for react-native, I have not found any documentation to make it work. Is there a solution available for react-native? Via the SDK or something else?

**Comment by joselcvarela** [comment]
Hey @HeinTouchableFR 
Try just open a webview with the link the SSO Provider
https://docs.directus.io/reference/authentication/#login-using-sso-providers

For example:
`https://directus.myserver.com/auth/login/google`

**Reply by HeinTouchableFR** [reply]
Opening the link in a web view does not work, error message from google. However, it is possible to open the link in a web browser thanks to the Linking component of react native. The connection works correctly, however, the user is redirected to the web version of the application, not to the mobile application. The directus SDK is therefore unable to recover the user's connection

**Reply by HeinTouchableFR** [reply]
However, I am able to retrieve a google authentication token thanks to the "@react-native-google-signin/google-signin" library. Is it possible to use the token returned during the connection thanks to this library directly with directus?

**Reply by joselcvarela** [reply]
What about using this technique?
https://stackoverflow.com/a/63169526/17931092

**Reply by HeinTouchableFR** [reply]
Great, it works perfectly. Thank you very much

**Reply by joselcvarela** [reply]
Glad it worked 🥳

**Comment by GustavoBlaze** [comment]
@HeinTouchableFR did you find a solution? i'm trying to do the same thing. If you had success, can you explain the steps you take?

**Reply by NilsBaumgartner1994** [reply]
Have a look at our PKCE solution?: https://github.com/directus/directus/discussions/23611
